wgpu/src/backend/direct.rs

```diff
@@ -927,7 +927,7 @@ impl crate::Context for Context {
                     defines: defines.clone(),
                 };
                 let mut parser = naga::front::glsl::Frontend::default();
-                let module = parser.parse(&options, shader).unwrap();
```

Comment:
wgpu's error handling story is a tad unusual because it implements WebGPU's error handling API and has to work in a completely asynchronous environment. So `create_shader_module` cannot return a `Result`.

Instead of unwrapping here, the error should be put into an error scope the same way it is done a few lines belowe with `handle_error`.

Reply:
Oh, I wondered why no `Result`s were being returned anywhere. I'll come back to this later today and make this change!